drop-transfer/src/service.rs

```diff
@@ -270,13 +271,13 @@ impl Service {
             {
                 Ok(res) => {
                     futures::future::join_all(
-                        res.events.iter().map(|ev| ev.stop_silent(Status::Canceled)),
+                        res.file_events
+                            .iter()
+                            .map(|ev| ev.stop_silent(Status::Canceled)),
                     )
                     .await;
 
-                    self.state
-                        .emit_event(crate::Event::OutgoingTransferCanceled(res.xfer, false));
-
+                    res.xfer_events.cancel(false).await;
```

Comment:
I am not 100% sure but I think if we would have one mutex over events(both files and transfers) it should, at least in theory simplfy and secure things :)

Reply:
I don't think it's that simple. In doing so we need to ensure all file events are flushed before canceling the transfer. Which might be very complex to ensure (remember that you can operate on files and transfer while offline).

Reply:
So imagine that you canceled the transfer while offline, thus emitting a `TransferCanceled` event and blocking further events. Meanwhile, your peer rejected some files. So he will see the files as being rejected and you will not because you blocked these events - blocked events mean no entries in DB as well.

Reply:
I think we misunderstood this one, I am saying that in this funciton body the file and event mutexes are separate ones and alos file mutex is locked/unlocked in each iteration, thus providing a window for a race condition. I am not saying anything about holding a mutex for whatever reason while we're offline :)

Reply:
I thought you were talking about blocking events after `TransferCanceled` not in the sense of holding mutex but in the sense of setting a `finished` flag or something.
If this is not the case, I don't see any problem with the current implementation. Nothing wrong should happen

Reply:
I actually don't even think the flag would be that bad. In your explained scenario with cancels and rejects, both are terminal states. And as for the mutex, I think it's still racy, why do you think nothing bad should happen?

Reply:
Events for transfer and files are not related IMHO. I gave you an example of why the flag would be bad with the offline peer. It will cause desync in what both peers state